src/core_init_atmosphere/mpas_init_atm_cases.F

```diff
@@ -5724,6 +5754,630 @@ subroutine init_atm_case_lbc(timestamp, block, mesh, nCells, nEdges, nVertLevels
+      !
+      call mpas_pool_get_array(mesh, 'fEdge', fEdge)
+      call mpas_pool_get_array(mesh, 'fVertex', fVertex)
+      call mpas_pool_get_array(mesh, 'latEdge', latEdge)
+      call mpas_pool_get_array(mesh, 'latVertex', latVertex)
+
+      do iEdge=1,nEdges
+         fEdge(iEdge)  = 2.0 * omega * sin(latEdge(iEdge))
+      end do
+      do iVtx=1,nVertices
+         fVertex(iVtx) = 2.0 * omega * sin(latVertex(iVtx))
+      end do
+
+
+      !
+      ! Compute weights used in advection and deformation calculation
+      !
+      call atm_initialize_advection_rk(mesh, nCells, nEdges, maxEdges, on_a_sphere, sphere_radius)
+      call atm_initialize_deformation_weights(mesh, nCells, on_a_sphere, sphere_radius)
+
+
+      !
+      ! Read PHIS field from cam_topo stream
+      !
+      if (MPAS_stream_mgr_stream_exists(stream_manager, 'cam_topo')) then
+         call MPAS_stream_mgr_read(stream_manager, 'cam_topo', rightNow=.true., whence=MPAS_STREAM_NEAREST, ierr=ierr)
+         if (ierr /= MPAS_STREAM_MGR_NOERR) then
+            call mpas_log_write('Error reading the ''cam_topo'' stream.', messageType=MPAS_LOG_CRIT)
+         end if
+      else
+         call mpas_log_write('')
+         call mpas_log_write('*************************************************************', messageType=MPAS_LOG_WARN)
+         call mpas_log_write('No ''cam_topo'' input stream with a PHIS field was defined.', messageType=MPAS_LOG_WARN)
+         call mpas_log_write('', messageType=MPAS_LOG_WARN)
+         call mpas_log_write('The terrain field will be set to zero everywhere. To specify', messageType=MPAS_LOG_WARN)
+         call mpas_log_write('a non-zero terrain field, define a ''cam_topo'' stream, e.g.,', messageType=MPAS_LOG_WARN)
+         call mpas_log_write('',                                    messageType=MPAS_LOG_WARN)
+         call mpas_log_write('<stream name="cam_topo"',             messageType=MPAS_LOG_WARN)
+         call mpas_log_write('        type="input"',                messageType=MPAS_LOG_WARN)
+         call mpas_log_write('        filename_template="topo.nc"', messageType=MPAS_LOG_WARN)
+         call mpas_log_write('        input_interval="none">',      messageType=MPAS_LOG_WARN)
+         call mpas_log_write('',                                    messageType=MPAS_LOG_WARN)
+         call mpas_log_write('    <var name="PHIS"/>',              messageType=MPAS_LOG_WARN)
+         call mpas_log_write('</stream>',                           messageType=MPAS_LOG_WARN)
+         call mpas_log_write('',                                    messageType=MPAS_LOG_WARN)
+         call mpas_log_write('*************************************************************', messageType=MPAS_LOG_WARN)
+         call mpas_log_write('')
+     end if
+
+
+      !
+      ! Set terrain field based on PHIS
+      !
+      call mpas_pool_get_array(mesh, 'ter', ter)
+      call mpas_pool_get_array(mesh, 'PHIS', PHIS)
+
```

Comment:
I'd request that we have an additional warning (Or perhaps an error?) that acts as a check that the file in the `cam_topo` stream is defined with `nCells` or not. 

When the `cam_topo` stream file is dimensioned with something other than `nCells`, a CAM-MPAS grid is produced with the `ter` being all zeros, and no warning is produced. This can be quite a perplexing to a user, as renaming a dimension in a file is not standard in any other MPAS workflows.

Thus, if a `cam_topo` stream is present *and* PHIS is all zero, we can assume that the `cam_topo` stream is not dimensioned by `nCells`, and thus either produce an error or a warning alerting the user that the ter field will be set to zero.

``` Fortran
if (MPAS_stream_mgr_stream_exists(stream_manager, 'cam_topo') .and. all(PHIS(:) == 0.0)) then
    ...
end if
```

Reply:
I do agree that the code should provide some indication to the user when the `PHIS` field could not be read from the "cam_topo" stream; however, I think the proper place to handle this would be the stream manager (or a lower level in the MPAS I/O stack). I was surprised to find in a test with `PHIS` dimensioned by `ncol` rather than by `nCells` that there were no warnings or errors at all from the I/O stack. I'll take a look in the I/O code to see how difficult it might be to catch the situation where the field to be read is dimensioned differently in the file compared with its `Registry.xml` definition. I would be fine with waiting to merge this PR until we can fix up the I/O code with a separate PR.

Reply:
PR #923 adds a critical error that will be triggered, e.g., when `PHIS` is dimensioned by `ncol` rather than `nCells` in the `cam_topo` stream.